vendor/github.com/spf13/cobra/bash_completions.md

```diff
@@ -6,6 +6,8 @@ Please refer to [Shell Completions](shell_completions.md) for details.
 
 For backward compatibility, Cobra still supports its legacy dynamic completion solution (described below).  Unlike the `ValidArgsFunction` solution, the legacy solution will only work for Bash shell-completion and not for other shells. This legacy solution can be used along-side `ValidArgsFunction` and `RegisterFlagCompletionFunc()`, as long as both solutions are not used for the same command.  This provides a path to gradually migrate from the legacy solution to the new solution.
 
+**Note**: Cobra's default `completion` command uses bash completion V2.  If you are currently using Cobra's legacy dynamic completion solution, you should not use the default `completion` command but continue using your own.
```

Comment:
Can you explain what this means? Maybe not in the doc but I don't know what it means and trying to understand.

Reply:
Cobra 1.2 automatically provides a `completion` command for programs that don't provide one themselves.  This greatly reduces the barrier-to-entry for providing completion support.  This "default" completion command uses the new bash completion v2 logic, also introduced in Cobra 1.2.  The v2 bash completion logic only supports custom completions written in Go (which kubectl started using through #96087 ).

But in any case, kubectl is not affected as it already defines its own `completion` command which still points to the bash completion v1 logic.

Please refer to https://github.com/spf13/cobra/blob/master/shell_completions.md#bash-completion-v2